components/resources/library/src/commonMain/kotlin/org/jetbrains/compose/resources/ImageResources.kt

```diff
@@ -56,10 +56,17 @@ private val emptyImageBitmap: ImageBitmap by lazy { ImageBitmap(1, 1) }
 @Composable
 fun imageResource(resource: DrawableResource): ImageBitmap {
     val resourceReader = LocalResourceReader.currentOrPreview
-    val imageBitmap by rememberResourceState(resource, resourceReader, { emptyImageBitmap }) { env ->
-        val path = resource.getResourceItemByEnvironment(env).path
-        val cached = loadImage(path, resourceReader) {
-            ImageCache.Bitmap(it.toImageBitmap())
+    val resourceEnvironment = rememberResourceEnvironment()
+    val imageBitmap by rememberResourceState(
+        resource, resourceReader, resourceEnvironment, { emptyImageBitmap }
+    ) { env ->
+        val item = resource.getResourceItemByEnvironment(env)
+        val resourceDensityQualifier = item.qualifiers.firstOrNull { it is DensityQualifier } as? DensityQualifier
+        val resourceDensity = resourceDensityQualifier?.dpi ?: DensityQualifier.MDPI.dpi
```

Comment:
Does Android do anything if the image just in `drawable` and device density is LDPI?

If it doesn't downscale, it should be `?: DensityQualifier.LDPI.dpi` (or better just `null`, with additional checks)

Reply:
On android the default density is MDPI. So, it works as expected:
<img width="317" alt="image" src="https://github.com/user-attachments/assets/79323732-3aa3-4492-9408-b6fa78a1deac">

Reply:
> On android the default density is MDPI

Is the `drawable` folder is for the default density MDPI?

I mean, from the user perspective, I would expect it to be density-independent, that requires to not do any downscale/upscale with images in it.

If `drawable` is really for MDPI in Android, let's keep the current behaviour.

Reply:
Yes. I see that android downscales png from the `drawable` folder on a LDPI device:
<img width="459" alt="image" src="https://github.com/user-attachments/assets/1952badc-ec0b-434e-b9e8-14ad631fbcac">
original image
<img width="372" alt="image" src="https://github.com/user-attachments/assets/06993fe5-2a56-4363-a725-97e32af7104d">

Reply:
we don't support nodpi yet.